### PR TITLE
add(swap):Integrated Half Button in Swap page

### PR DESCRIPTION
**Pull request by gsantini-ube**
-Integrated Half button and Half button logic

Added button, buttong-group for buttons

image
image
image

### CODE REVIEW

File: src/pages/Swap/index.tsx [hunk 1]
```diff
@@ -273,6 +276,16 @@ export default function Swap() {
     }
   }, [maxAmountInput, onUserInput, currencies, chainId])
 
+  const handleHalfInput = useCallback(() => {
+    if (maxAmountInput) {
```
Comment (aaronmgdr) [inline src/pages/Swap/index.tsx:280]: why this maxAmount input conditional?

File: src/pages/Swap/index.tsx [hunk 2]
```diff
@@ -273,6 +276,16 @@ export default function Swap() {
...
+      if (currencies?.INPUT?.address === CELO[chainId].address) {
+        onUserInput(Field.INPUT, Math.max(Number(maxAmountInput.toExact()) * 0.5, 0).toString())
+      } else {
+        onUserInput(Field.INPUT, (Number(maxAmountInput.toExact()) * 0.5).toString())
```
Comment (aaronmgdr) [inline src/pages/Swap/index.tsx:284]: For half amount it doesnt matter if its CELO token or not

**Review by aaronmgdr — CHANGES_REQUESTED** [review]
Can we do this without adding 2 new packages ? I think so. 

Fixed. Thanks

**Comment by aaronmgdr** [comment]
I think this takes up too much space on mobile but that it also would not be good for it to be smaller as buttons are already pretty small. 

I know that's not great feedback. Im not sure the best solution.

IMG_1BDBE45EA24E-1

**Comment by BriungRi** [comment]
Here's a different design pattern we could adopt.
Screen Shot 2021-12-10 at 6 36 14 PM

Place 4 buttons at the bottom for [25% 50% 75% 100%]

**Comment by gsantini-ube** [comment]
I have changed the layout of panel.
image
image
image
image

**Comment by BriungRi** [comment]
Can we have some additional margin right between the amount number and the 50% button?

**Review by BriungRi — APPROVED** [review]

**Review by aaronmgdr — APPROVED** [review]
I agree with Brian, needs more space between the two buttons so they can be pressed by fingers

**Comment by gsantini-ube** [comment]
I changed additional margin between number input and buttons.

image
image

**Pull request merged**